src/book.py

```diff
@@ -574,73 +577,79 @@ def _read_kraken_ledgers(self, file_path: Path) -> None:
                 # in the public trade history and are skipped.
                 # For staking / unstaking / staking reward actions, deposits /
                 # withdrawals only occur once and will be ignored.
+                # The "appended" flag stores if an operation for a given refid has
+                # already been appended to the operations list:
+                # == None: Initial value, this is the first occurence
+                # == False: No operation has been appended, this is the second occurene
+                # == True: Operation has already been appended, this should not happen
                 if operation in ["Deposit", "Withdrawal"]:
-                    # search for refid in refids list
-                    refid_idxs = [
-                        idx
-                        for idx, op in enumerate(held_operations)
-                        if op["refid"] == refid
-                    ]
-                    # refid should not exist more than once in list
-                    if len(refid_idxs) > 1:
-                        log.error(
-                            f"{file_path} row {row}: More than two entries with refid "
-                            f"{refid} should not exist ({operation}). "
-                            "Please create an Issue or PR."
-                        )
-                        raise RuntimeError
-                    # if refid already exists, assert that data of operations agree
-                    if refid_idxs:
-                        idx = refid_idxs[0]
+                    # If this is the first occurence, set the "appended" flag to false
+                    # and don't append the operation to the list. Instead, store the
+                    # data for verifying or appending it later.
+                    if self.kraken_held_ops[refid]["appended"] is None:
+                        append_operation = False
+                        self.kraken_held_ops[refid]["appended"] = False
+                        self.kraken_held_ops[refid]["operation"] = operation
```

Comment:
If you save all operation data as one key e.g. `d[refId]["operation"] : tuple or dict` you can easily discard the data as soon as we find the second value too free some memory. (~#overoptimization :D)